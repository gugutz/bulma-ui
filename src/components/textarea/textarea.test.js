import React from 'react'
import renderer from 'react-test-renderer'
import Textarea from './textarea'

describe('<Textarea /> component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Textarea>some content</Textarea>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  const types = ['text', 'password', 'email', 'tel']

  for (const type of types) {
    it('should support type prop with ' + type, () => {
      const tree = renderer
        .create(<Textarea type={type}>some content</Textarea>)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  it('should support disabled prop', () => {
    const tree = renderer
      .create(<Textarea disabled>some content</Textarea>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support readOnly prop', () => {
    const tree = renderer
      .create(<Textarea readOnly>some content</Textarea>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support isStatic prop', () => {
    const tree = renderer
      .create(<Textarea isStatic>some content</Textarea>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support other props from textarea', () => {
    const tree = renderer
      .create(
        <Textarea name="email" placeholder="email@example.com">
          some content
        </Textarea>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  const colors = ['primary', 'info', 'success', 'warning', 'danger', 'default']

  for (const color of colors) {
    it('should support color prop with ' + color, () => {
      const tree = renderer
        .create(<Textarea color={color}>some content</Textarea>)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  const states = ['hovered', 'focused', 'normal']

  for (const state of states) {
    it('should support state prop with ' + state, () => {
      const tree = renderer
        .create(<Textarea state={state}>some content</Textarea>)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  const sizes = ['small', 'normal', 'medium', 'large']

  for (const size of sizes) {
    it('should support size prop with ' + size, () => {
      const tree = renderer
        .create(<Textarea size={size}>some content</Textarea>)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
})
