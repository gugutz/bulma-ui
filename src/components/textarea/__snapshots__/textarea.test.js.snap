// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> component should render correctly 1`] = `
<textarea
  className="textarea"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support color prop with danger 1`] = `
<textarea
  className="textarea is-danger"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support color prop with default 1`] = `
<textarea
  className="textarea"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support color prop with info 1`] = `
<textarea
  className="textarea is-info"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support color prop with primary 1`] = `
<textarea
  className="textarea is-primary"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support color prop with success 1`] = `
<textarea
  className="textarea is-success"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support color prop with warning 1`] = `
<textarea
  className="textarea is-warning"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support disabled prop 1`] = `
<textarea
  className="textarea"
  disabled={true}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support isStatic prop 1`] = `
<textarea
  className="textarea is-static"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support other props from textarea 1`] = `
<textarea
  className="textarea"
  disabled={false}
  name="email"
  placeholder="email@example.com"
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support readOnly prop 1`] = `
<textarea
  className="textarea"
  disabled={false}
  readOnly={true}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support size prop with large 1`] = `
<textarea
  className="textarea is-large"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support size prop with medium 1`] = `
<textarea
  className="textarea is-medium"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support size prop with normal 1`] = `
<textarea
  className="textarea"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support size prop with small 1`] = `
<textarea
  className="textarea is-small"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support state prop with focused 1`] = `
<textarea
  className="textarea is-focused"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support state prop with hovered 1`] = `
<textarea
  className="textarea is-hovered"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support state prop with normal 1`] = `
<textarea
  className="textarea"
  disabled={false}
  readOnly={false}
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support type prop with email 1`] = `
<textarea
  className="textarea"
  disabled={false}
  readOnly={false}
  type="email"
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support type prop with password 1`] = `
<textarea
  className="textarea"
  disabled={false}
  readOnly={false}
  type="password"
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support type prop with tel 1`] = `
<textarea
  className="textarea"
  disabled={false}
  readOnly={false}
  type="tel"
>
  some content
</textarea>
`;

exports[`<Textarea /> component should support type prop with text 1`] = `
<textarea
  className="textarea"
  disabled={false}
  readOnly={false}
  type="text"
>
  some content
</textarea>
`;
